--- CPSTestCase.py	2004-08-05 14:55:05.000000000 +0200
+++ CPSTestCase.py-patched	2004-08-05 14:50:41.000000000 +0200
@@ -46,6 +46,11 @@
 try: ZopeTestCase.installProduct('CPSMailBoxer', quiet=1)
 except: pass
 
+test_cpsskins = (os.environ.get('CPSSKINS_TARGET', '') == 'CPS3')
+if test_cpsskins:
+    try: ZopeTestCase.installProduct('CPSSkins', quiet=1)
+    except: pass
+
 from AccessControl.SecurityManagement \
     import newSecurityManager, noSecurityManager
 
@@ -163,6 +168,8 @@
         self.login()
         self.addPortal(portal_id)
         self.fixupTranslationServices(portal_id)
+        if test_cpsskins:
+            self.setupCPSSkins(portal_id)
         self.logout()
 
     def addUser(self):
@@ -188,6 +195,12 @@
         for domain in localizer.objectIds():
             setattr(localizer, domain, DummyMessageCatalog())
 
+    def setupCPSSkins(self, portal_id):
+        portal = getattr(self.app, portal_id)
+        factory = portal.manage_addProduct['CPSSkins']
+        factory.manage_addCPSSkins(portal_id, SourceSkin='Basic', \
+             Target='CPS3', ReinstallDefaultThemes=1)
+
     def logout(self):
         noSecurityManager()
         get_transaction().commit()
